File: Ordenamiento/StoogeSort.java
public class StoogeSort{

	int i,n;
	int[] arr = {50,17,48,23,8,60};

	public static void main(String[] args){
		StoogeSort ss = new StoogeSort();
		ss.stoogeSort(ss.arr, 0, ss.arr.length-1);  //Ordenamiento StoogeSort
		ss.n = ss.arr.length;

		for(ss.i=0; ss.i<=ss.n-1; ss.i++){
			System.out.print(ss.arr[ss.i] + ", ");
		}
		System.out.println("\n");
	}

	public void stoogeSort(int[] A, int inf, int sup){
		if (A[inf] > A[sup]){
			// Intercambio
			int temp = A[sup];
			A[sup] = A[inf];
			A[inf] = temp;
		}

		// Si el subarreglo tiene más de dos elementos, se ordenan por tercios
		if ((sup-inf+1) > 2){
			int t = (sup-inf+1)/3;
			stoogeSort(A,inf,sup-t); // Se ordenan los primeros dos tercios del arreglo
			stoogeSort(A,inf+t,sup); // Se ordenan los ultimos dos tercios del arreglo
			stoogeSort(A,inf,sup-t); // Se vuelven a ordenar los primeros dos tercios del arreglo
		}
	}

}

File: EDAII x Python/Ordenamiento/StoogeSort.py
class StoogeSort:

	arr = []	# Arreglo de elementos

	def __init__(self, arr):
		self.arr = arr

	def stoogeSort(self, arr, inf, sup):
		'Algoritmo de ordenamiento.'
		if arr[inf] > arr[sup]:
			# Intercambio
			temp = arr[sup]
			arr[sup] = arr[inf]
			arr[inf] = temp

		# Si el subarreglo tiene más de dos elementos, se ordenan por tercios
		if (sup-inf+1) > 2:
			t = int((sup-inf+1)/3)
			self.stoogeSort(arr,inf,sup-t)	# Se ordenan los primeros dos tercios del arreglo
			self.stoogeSort(arr,inf+t,sup)	# Se ordenan los ultimos dos tercios del arreglo
			self.stoogeSort(arr,inf,sup-t)	# Se vuelven a ordenar los primeros dos tercios del arreglo

# Ejecutando objetos
programa = StoogeSort([50,17,48,23,8,60])
programa.stoogeSort(programa.arr,0,len(programa.arr)-1)
print(programa.arr)